lametro/models.py

```diff
@@ -189,12 +192,68 @@ def get_queryset(self):
         return super().get_queryset()
 
 
-class LAMetroEvent(Event):
+class LiveMediaMixin(object):
+    BASE_MEDIA_URL = 'http://metro.granicus.com/mediaplayer.php?'
+    GENERIC_ENGLISH_MEDIA_URL = BASE_MEDIA_URL + 'camera_id=3'
+    GENERIC_SPANISH_MEDIA_URL = BASE_MEDIA_URL + 'camera_id=2'
+
+
+    @property
+    def bilingual(self):
+        '''
+        Upstream, when an English-language event can be paired with a Spanish-
+        language event, the GUID of the Spanish-language event is added to the
+        extras dictionary, e.g., return True if the sap_guid is present, else
+        return False.
+        '''
+        return bool(self.extras.get('sap_guid'))
+
+
+    def _valid_or_generic(self, media_url, fallback):
+        '''
+        Return the given URL if it returns a 200 status code, else return
+        generic fallback.
+        '''
+        response = requests.get(media_url)
+
+        if response.ok and 'The event you selected is not currently in progress' not in response.text:
+            return media_url
+        else:
+            return fallback
+
+
+    @property
+    def english_live_media_url(self):
+        guid = self.extras['guid']
+        english_url = self.BASE_MEDIA_URL + 'event_id={guid}'.format(guid=guid)
+
+        return self._valid_or_generic(english_url,
```

Comment:
For me this would be clearer if you _valid_or_gener was just _valid, and then you handled the fallback case here.